x-pack/plugins/fleet/public/components/new_enrollment_key_modal.tsx

```diff
@@ -65,7 +78,11 @@ export const NewEnrollmentTokenModal: React.FunctionComponent<Props> = ({
   const { notifications } = useStartServices();
   const policyIdDefaultValue = agentPolicies.find((agentPolicy) => agentPolicy.is_default)?.id;
   const form = useCreateApiKeyForm(
-    policyIdDefaultValue,
+    policyIdDefaultValue
+      ? policyIdDefaultValue
+      : agentPolicies.length > 0
+      ? agentPolicies[0].id
+      : undefined,
```

Comment:
good catch! something is weird though, if I comment out this code (line 81-85) and leave only undefined, it means there is no default value, and the logic should take the policy id of the selected policy, isn't it?

that doesn't seem to happen, initially there is a policy selected in the dropdown (might not even be the first one), but the input value is not set to that policy's id, and I'm getting the validation error:
![image](https://user-images.githubusercontent.com/90178898/150096665-d01d36bf-dd90-4094-b053-0247f3e9d3de.png)

Reply:
I think we could have a bug in the hooks I wrote for the input if you pass undefined I think the defaultValue will be an empty string `""`

Reply:
I see, so when removing the default policies, we can keep the default as the first policy (pre-selected in dropdown)